lib/manager/cocoapods/readme.md

```diff
@@ -0,0 +1,232 @@
+
+No
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+Package file uses ruby syntax.
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+Using regex
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+It distinguishes dependencies for different targets (testing target, target for TV OS etc)
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+Dependency indicated by keyword `pod` and name with version.
+
+Also, it's possible to use a local podspec file on the machine, or in the root of library repo.
+In case of the repo you also can specify branch, tag or commit.
+
+```
+pod 'AFNetworking', '~> 2.6'
+pod 'Alamofire', :path => '~/Documents/Alamofire'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :branch => 'dev'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :tag => '3.1.1'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :commit => '0f506b1c45'
+```
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+Cocoapods supports dependencies from git source and local path.
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+Package file use semver 2.0. ([details](https://semver.org))
+
+---
+
+#### Does this versioning scheme support range constraints, e.g. `^1.0.0` or `1.x`?
+
+Versioning scheme supports logical operators:
+
+- `> 0.1` Any version higher than 0.1
+- `>= 0.1` Version 0.1 and any higher version
+- `< 0.1` Any version lower than 0.1
+- `<= 0.1` Version 0.1 and any lower version
+
+In addition to the logic operators CocoaPods has an optimistic operator `~>`:
+
+- `~> 0.1.2` Version 0.1.2 and the versions up to 0.2, not including 0.2 and higher
+- `~> 0.1` Version 0.1 and the versions up to 1.0, not including 1.0 and higher
+- `~> 0` Version 0 and higher, this is basically the same as not having it.
+
+---
+
+#### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
+
+Package manager used for applications only
+
+---
+
+#### If ranges are supported, are there any cases when Renovate should pin ranges to exact versions if rangeStrategy=auto?
+
+I don't see such reason.
+
+## Lookup
+
+#### Is a new datasource required? Provide details
+
+There is a index service for cocoapods. Developers always search for pod on this [site](https://github.com/CocoaPods):
+
+Also, all podspec are stored in one [repo](https://github.com/CocoaPods/Specs)
```

Comment:
Wow, that looks like a different approach to anything I've ever seen. We'll need to discuss more about how to do that efficiently, e.g. cloning it, caching it, etc.